### PR TITLE
(VANAGON-157) Make curl be more clear about fetch errors.

### PR DESCRIPTION
**Pull request by e-gris**
This commit adds the '--silent' '--show-error' and '--fail' flags to
calls made by vanagon to force bad fetches to return non-zero exit
status in the shell.

As a side-effect, some of the curl output statistics is silenced. I
think this silencing is desirable.

The previous behavior looked like this upon success:

$ curl -o STUFF https://artifactory.delivery.puppetlabs.net/artifactory/scratchpad__local/exg-test/foo.txt
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    29  100    29    0     0    159      0 --:--:-- --:--:-- --:--:--   160
$ echo $?
0

The previous behavior looked like this upon  failure:

$ curl -o STUFF https://artifactory.delivery.puppetlabs.net/artifactory/scratchpad__local/exg-test/foo.txt.BAD
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    80    0    80    0     0    419      0 --:--:-- --:--:-- --:--:--   421
$ echo $?
0

The new behavior looks like this upon success:

$ curl --silent --show-error --fail -o STUFF https://artifactory.delivery.puppetlabs.net/artifactory/scratchpad__local/exg-test/foo.txt
$ echo $?
0

The new behavior looks like this upon failure:

$ curl --silent --show-error --fail -o STUFF https://artifactory.delivery.puppetlabs.net/artifactory/scratchpad__local/exg-test/foo.txt.BAD
curl: (22) The requested URL returned error: 404 Not Found
$ echo $?
22

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This changelog adheres to [Keep a CHANGELOG](http://keepachangelog.com/).
 
 ## [Unreleased]
+### Changed
+- Make curl calls fail on error
```
Comment (mwaggett) [inline CHANGELOG.md]: Can you reference the ticket this is resolving?

File: spec/lib/vanagon/platform/dsl_spec.rb [hunk 2]
```diff
@@ -58,33 +66,31 @@
...
 
     it "works for specifically redhat platforms" do
       plat = Vanagon::Platform::DSL.new('redhat-7-fixture')
       expect(SecureRandom).to receive(:hex).and_return(hex_value)
       plat.instance_eval(redhat_7_platform_block)
       plat.yum_repo(el_definition)
-      expect(plat._platform.provisioning).to include("curl -o '/etc/yum.repos.d/#{hex_value}-pl-puppet-agent-0.2.1-el-7-x86_64.repo' '#{el_definition}'")
-    end
-
-    # This test currently covers wrlinux 5 and 7
```
Comment (mwaggett) [inline spec/lib/vanagon/platform/dsl_spec.rb]: Can the removal of this test go in a separate commit? It seems pretty unrelated to the curl work.

Comment (e-gris) [reply spec/lib/vanagon/platform/dsl_spec.rb]: Restored the code in this branch.

Comment (mwaggett) [reply spec/lib/vanagon/platform/dsl_spec.rb]: I didn't mean that you should put it back, just make the change in a separate commit. but this is fine too.

**Comment by mwaggett** [comment]
holding off on re-reviewing until ^checks have passed

File: spec/lib/vanagon/platform/dsl_spec.rb (hunk 2 above)
Comment (e-gris) [reply spec/lib/vanagon/platform/dsl_spec.rb]: Reverting it was the better choice. I looked further and a correct, complete removal of this platform, even just from the rspec files, would involve more changes. I'll file another ticket to do that.
